docs/docs-beta/docs/guides/deployment/kubernetes.md

```diff
@@ -1,4 +1,221 @@
+
+### Step 2.2: Build and push a Docker image
+
+To build your Docker image, run the following command from the directory where your Dockerfile is located:
+
+```bash
+docker build . -t iris_analysis:1
+```
+This builds the Docker image from Step 2.1 and gives it the name `iris_analysis` and tag `1`. You can set custom values for both the name and the tag. We recommend that each time you rebuild your Docker image, you assign a new value for the tag to ensure that the correct image is used when running your code.
+
+
+If you are using a Docker image registry, push the image to your registery. If you are following along on your local machine, you can skip this command.
+
+```bash
+docker push iris_analysis:1
+```
+
+TODO - specific instructions if pushing to an image registry?
+
+
+## Step 3: Configure `kubectl` to point at a Kubernetes cluster
+Before you can deploy Dagster, you need to configure `kubectl` to develop against the Kubernetes cluster where you want Dagster to be deployed.
+
+If you are using Docker Desktop and the included Kubernetes server, you will need to create a context first. If you already have a Kubernetes cluster and context created for your Dagster deployment you can skip running this command.
+```bash
+kubectl config set-context dagster --namespace default --cluster docker-desktop --user=docker-desktop
+```
+
+Ensure that `kubectl` is using the correct context by running:
+```bash
+kubectl config use-context <context-name>
+```
+Where `<context-name>` is the name of the context you want to use. For example, if you ran the `kubectl config set-context` command above, you will run
+```bash
+kubectl config use-context dagster
+```
+
+## Step 4: Add the Dagster Helm chart repository
+
+Dagster publishes [Helm charts](https://artifacthub.io/packages/helm/dagster/dagster) for deploying Dagster. New Helm charts are published for each version of Dagster. You should use the Helm chart version that matches the version of Dagster you have installed.
+
+To install the Dagster Helm charts, run the following command:
+
+```bash
+helm repo add dagster https://dagster-io.github.io/helm
+```
+
+If you have previously added the Dagster Helm charts, run the following command to update the repository:
+
+```bash
+helm repo update
+```
+
+## Step 5: Configure the Helm chart for your deployment
+
+You will need to modify some values in Dagster's Helm chart to deploy your Dagster project.
+
+### Step 5.1: Copy the default Helm chart values into a `values.yaml`
+
+Run the following command to copy the values installed from the published Helm charts so that you can modify them:
+
+```bash
+helm show values dagster/dagster > values.yaml
+```
+TODO - where to copy this file to?
+
+### Step 5.2: Modify the `values.yaml` file for your deployment
+The `values.yaml` file contains configuration options you can set for your deployment. There are comments in the `values.yaml` file explaining these options, and you can learn more about them [here](/todo).
```

Comment:
Not sure if a doc exists that documents the various options you can set in the `values.yaml`

Reply:
I think I saw a post where Jordan found a way to generate them. For now a /todo is fine

Reply:
https://artifacthub.io/packages/helm/dagster/dagster?modal=values has all the possible values.

The thing I was generating was documentation around dagster.yaml config (and in my case, only for Dagster+ - although we could probably extend it to use elsewhere in the docs site). Similar, but not quite the same:

- Helm values customize our Helm templates. Our Helm templates describe various Kubernetes resources to be created with the Kubernetes API (deployments, services, configmaps, etc.)
- dagster.yaml configures the `DagsterInstance` class.
- This Helm template creates a [configmap](https://kubernetes.io/docs/concepts/configuration/configmap/) that [contains a dagster.yaml](https://github.com/dagster-io/dagster/blob/1c47efad19fd3027704fc4f5d8a7c9da18bdb699/helm/dagster/templates/configmap-instance.yaml#L11). So some of the options in this particular values.yaml map onto options in a dagster.yaml that it creates for us.

But yeah, I'd just link directly out to the Helm template docs in artifacthub here.